### PR TITLE
Adopting new pmem library

### PR DESCRIPTION
**Pull request by harish-24**
Patch adopts new pmem library which is moved to avocado framework so that other tests can also leverage this.

Signed-off-by: Harish <harish@linux.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
Depends on https://github.com/avocado-framework/avocado/pull/3568

**Comment by harish-24** [comment]
```
 (01/20) memory/ndctl.py:NdctlTest.test_bus_ids: PASS (0.28 s)
 (02/20) memory/ndctl.py:NdctlTest.test_dimms: PASS (0.28 s)
 (03/20) memory/ndctl.py:NdctlTest.test_regions: PASS (0.31 s)
 (04/20) memory/ndctl.py:NdctlTest.test_namespace: PASS (39.25 s)
 (05/20) memory/ndctl.py:NdctlTest.test_disable_enable_ns: FAIL: PMemException('Namespace enable failed for "namespace1.0"',) (1.31 s)
 (06/20) memory/ndctl.py:NdctlTest.test_namespace_modes: PASS (44.43 s)
 (07/20) memory/ndctl.py:NdctlTest.test_namespace_devmap: PASS (0.78 s)
 (08/20) memory/ndctl.py:NdctlTest.test_multiple_namespaces_region: PASS (24.92 s)
 (09/20) memory/ndctl.py:NdctlTest.test_multiple_ns_multiple_region: PASS (48.54 s)
 (10/20) memory/ndctl.py:NdctlTest.test_multiple_ns_modes_region: PASS (1.26 s)
 (11/20) memory/ndctl.py:NdctlTest.test_nslot_namespace: PASS (25.27 s)
 (12/20) memory/ndctl.py:NdctlTest.test_namespace_reconfigure: PASS (44.37 s)
 (13/20) memory/ndctl.py:NdctlTest.test_check_namespace: PASS (1.46 s)
 (14/20) memory/ndctl.py:NdctlTest.test_check_numa: PASS (0.84 s)
 (15/20) memory/ndctl.py:NdctlTest.test_label_read_write: FAIL: Created namespace not found after label restore (22.49 s)
 (16/20) memory/ndctl.py:NdctlTest.test_daxctl_list: PASS (22.40 s)
 (17/20) memory/ndctl.py:NdctlTest.test_sector_write: \^PASS (561.86 s)
 (18/20) memory/ndctl.py:NdctlTest.test_fsdax_write: PASS (159.73 s)
 (19/20) memory/ndctl.py:NdctlTest.test_map_sync: PASS (23.25 s)
 (20/20) memory/ndctl.py:NdctlTest.test_devdax_write: PASS (825.47 s)
```

**Comment by harish-24** [comment]
@PraveenPenguin Please help in reviewing this

**Comment by PraveenPenguin** [comment]
> @PraveenPenguin Please help in reviewing this

@harish-24 yes , sorry for late review will close this as soon as possible

File: memory/ndctl.py [hunk 1]
```diff
@@ -309,103 +142,113 @@ def setUp(self):
...
             self.daxctl = 'daxctl'
 
         self.opt_dict = {'-B': 'provider',
                          '-D': 'dev', '-R': 'dev', '-N': 'dev'}
         self.modes = ['raw', 'sector', 'fsdax', 'devdax']
         self.part = None
         self.disk = None
+        self.plib = pmem.PMem(self.ndctl, self.daxctl)
+        if not self.plib.check_buses():
+            self.cancel("Test needs atleast one region")
```
Comment (PraveenPenguin) [inline memory/ndctl.py]: ok

**Pull request merged**